src/lib/systemd_journal/entries_dialog.rb

```diff
@@ -28,11 +28,12 @@
 module SystemdJournal
   # Dialog to display journal entries with several filtering options
   class EntriesDialog < UI::Dialog
-    def initialize
-      super
+    # @param query [QueryPresenter] optional initial query
+    def initialize(query: nil)
```

Comment:
This is something that was probably a little bit weird already in the original implementation, receiving a `QueryPresenter` instead of a `Query` as argument.

It probably makes more sense to receive a `Query` and create the `QueryPresenter` to wrap it as part of `EntriesDialog#initialize` (which probably means touching the `QueryPresenter` API, as well, I know).